### ISSUE TITLE
Create a function app to power off AKS nodes (VMSS) outside of core dev hours 6-9am

### ISSUE THREAD

**Issue by adamrushuk** [adamrushuk/aks-nexus-velero#43]
I usually dev between 06:00 - 09:00, so would like the AKS nodes to be deallocated if I forget to do this manually.

- [x] Test PowerShell 7 func app worker: https://github.com/Azure/azure-functions-powershell-worker/issues/371#issuecomment-641026259 - just add `$env:FUNCTIONS_WORKER_RUNTIME_VERSION = '~7'` when local, or add application setting `FUNCTIONS_WORKER_RUNTIME_VERSION = '~7'` for hosted function apps.
- [x] Run `choco install azure-functions-core-tools-3 --params "'/x64'"` for local testing.
- [x] Add support for UK time comparison
- [x] Add zip script and zip hash script to find replace token in TF var 
- [x] Deallocate: mon-fri 06:00 - 09:00.
- [ ] Deallocate: sat-sun 15:00 - 06:00.
- [ ] Only notify (push notification to mobile) that AKS nodes running: sat-sun 15:00 - 06:00 https://www.dennisrye.com/post/twitchscanner/ 

Use locks or tags to skip prod VMs

https://docs.microsoft.com/en-us/azure/azure-functions/functions-create-first-function-vs-code

Use config appsettings for Service Principal auth during local testing, and MSI once deployed. https://github.com/Azure/Azure-Functions/issues/644#issuecomment-356088965

## Reference

- https://markheath.net/post/managed-identity-key-vault-azure-functions
- https://www.terraform.io/docs/providers/azurerm/r/function_app.html
- https://github.com/JustinGrote/terraform-azurerm-azure-function-powershell/blob/master/main.tf

**Comment by adamrushuk** [comment]
https://github.com/Azure/azure-functions-core-tools#notice-to-debug-functions-under-vscode-x64-bitness-is-required

**Comment by adamrushuk** [comment]
Local debugging not working atm, but a new version of runtime with PS7 is inbound, so manually testing in Portal for now.

**Comment by adamrushuk** [comment]
Code now working from manual changes developing within browser.
Now need to deploy function app via Terraform.

**Comment by kavyabgowda** [comment]
Is this possible working from Terraform? I created a powershell azure function app with timer trigger functions and used az modules inside the site folder and referenced in the code...when I do manual upload of the code through storage explorer or kudu it works. But if I use terraform Website_Use_Zip the code is not working just printing the information logs nothing else. I got to know that .azure folder is not getting created because the code is uploaded to blob. Is there any to make it work with terraform completely without manually doing.
